tests/testapp/urls.py

```diff
@@ -37,4 +37,14 @@
     path(
         'owner/shelf', views.shelf_list_owner_view, name='test_shelf_list_owner_view',
     ),
+    path(
+        'owner/shelf/<str:shelf_id>/books',
```

Comment:
there's a missing trailing slash here

Reply:
All other paths in this file don't have a trailing slash, should I just add it to all?